string.go

```diff
@@ -30,40 +28,45 @@ import (
 	"unicode/utf8"
 )
 
-// AESEncrypt encrypts text and given key with AES.
-func AESEncrypt(key, text []byte) ([]byte, error) {
+// AESEncrypt encrypts plaintext with the given key and nonce using AES in GCM mode.
+func AESEncrypt(key, nonce, plaintext []byte) ([]byte, error) {
 	block, err := aes.NewCipher(key)
 	if err != nil {
 		return nil, err
 	}
-	b := base64.StdEncoding.EncodeToString(text)
-	ciphertext := make([]byte, aes.BlockSize+len(b))
-	iv := ciphertext[:aes.BlockSize]
-	if _, err := io.ReadFull(rand.Reader, iv); err != nil {
+
+	b := base64.StdEncoding.EncodeToString(plaintext)
```

Comment:
Why do you encrypt the base64-encoded string representation of `plaintext` and not `plaintext` directly?
Base64-encoding creates a lot of unnecessary overhead.